src/main.rs

```diff
@@ -18,143 +18,119 @@
-		let (call, mmr_store) = murmur::create(
+		let (payload, store) = murmur::create(
 			username.into(),
 			seed.into(),
 			get_ephem_msk(), // TODO: replace with an hkdf? https://github.com/ideal-lab5/murmur/issues/13
 			schedule,
 			round_pubkey_bytes,
 		)
-		.map_err(|_| Status::InternalServerError)?;
+		.map_err(|e| (Status::InternalServerError, MurmurError(e).to_string()))?;
 		// 3. add to storage
-		store::write(mmr_store.clone());
-		// sign and send the call
-		let from = dev::alice();
-		let _events = client
-			.tx()
-			.sign_and_submit_then_watch_default(&call, &from)
-			.await
-			.map_err(|_| Status::InternalServerError)?;
-		Ok("MMR proxy account creation successful!".to_string())
+		store::write(store.clone());
+		// 4. return the call data
+		Ok(CreateResponse { payload: payload.into() })
 	})
 	.await
 }
 
 #[post("/execute", data = "<request>")]
 /// Execute a transaction from the wallet
-async fn execute(cookies: &CookieJar<'_>, request: Json<ExecuteRequest>) -> Result<String, Status> {
+async fn execute(
+	cookies: &CookieJar<'_>,
+	request: Json<ExecuteRequest>,
+) -> Result<ExecuteResponse, (Status, String)> {
 	check_cookie(cookies, |username, seed| async {
-		let (client, current_block_number, _) =
-			murmur::idn_connect().await.map_err(|_| Status::InternalServerError)?;
-
-		let from_ss58 = sp_core::crypto::AccountId32::from_ss58check(&request.to).unwrap();
-
-		let bytes: &[u8] = from_ss58.as_ref();
-		let from_ss58_sized: [u8; 32] = bytes.try_into().unwrap();
-		let to = AccountId32::from(from_ss58_sized);
-		let balance_transfer_call = Balances(Call::transfer_allow_death {
-			dest: MultiAddress::<_, u32>::from(to),
-			value: request.amount,
-		});
-
 		let store = store::load();
-		let target_block_number = current_block_number + 1;
+		let target_block = request.current_block + 1;
 
-		let tx = murmur::prepare_execute(
+		let runtime_call = RuntimeCall::decode(&mut &request.runtime_call[..])
+			.map_err(|e| (Status::InternalServerError, e.to_string()))?;
+
+		let payload = murmur::prepare_execute(
 			username.into(),
 			seed.into(),
-			target_block_number,
+			target_block,
 			store,
-			balance_transfer_call,
+			runtime_call,
 		)
-		.map_err(|_| Status::InternalServerError)?;
-
-		// submit the tx using alice to sign it
-		let _ = client.tx().sign_and_submit_then_watch_default(&tx, &dev::alice()).await;
+		.map_err(|e| (Status::InternalServerError, MurmurError(e).to_string()))?;
 
-		Ok("Transaction executed".to_string())
+		Ok(ExecuteResponse { payload: payload.into() })
 	})
 	.await
 }
 
 #[launch]
 fn rocket() -> _ {
-	rocket::build().mount("/", routes![login, new, execute])
+	let cors = CorsOptions::default()
+		.allowed_origins(AllowedOrigins::all())
+		.allowed_methods(
+			vec![Method::Get, Method::Post, Method::Patch]
+				.into_iter()
+				.map(From::from)
+				.collect(),
+		)
+		.allowed_headers(AllowedHeaders::all())
+		.allow_credentials(true)
+		.to_cors()
+		.unwrap();
```

Comment:
just making sure we're comfortable with this unwrap. I think it should be fine since it would only fail on startup, but we could also handle it if needed

Reply:
This is just fine, as you said it would only fail at startup